index.js

```diff
@@ -7,13 +7,17 @@ var getGitInfo = require('git-repo-info');
 
 module.exports = function version(shaLength, root) {
   var projectPath = root || process.cwd();
-
+  var packageVersion  = require(path.join(projectPath, 'package.json')).version;
   var info = getGitInfo(projectPath);
+
   if (info.tag) {
-    return info.tag;
+    if (packageVersion && info.tag.includes(packageVersion)) {
```

Comment:
Do you like to support node.js 0.12? In node 0.12 method was named `contains`. I guess to support older version a fallback to `.indexOf(substring) !== -1` would be needed.